Create new Docker dev image

am97 opened the pull request:
Current Docker development image has not been updated in a while, so I got some issues when using it:
- Related to PostgreSQL (because extensions `pg_trgm` and `unaccent` are not created)
- Related to Redis (because it is not installed)
- Related to server on port 9000

The first two are easy to solve, but I was unable to fix the third with the current dev image. (more on this later)

So I created a new Docker dev image, using some lines from the current one, and trying to follow closely the [dependencies guide for Ubuntu](https://github.com/Chocobozzz/PeerTube/blob/develop/support/doc/dependencies.md#debian--ubuntu-and-derivatives). It needs review, as there is still one error.

### Explanation of the issues:
#### Steps I took with current Docker dev image:
```
docker pull chocobozzz/peertube-dev
docker run -d -p 3000:3000 -p 9000:9000 --name peertube chocobozzz/peertube-dev
docker exec -it peertube bash
(in container) sudo -u postgres psql -c "CREATE EXTENSION pg_trgm;" peertube_dev
(in container) sudo -u postgres psql -c "CREATE EXTENSION unaccent;" peertube_dev
(in container) npm run dev
```
From the host:
In `localhost:3000`, the web UI is working, but displays errors (`Error getting about from server`)
In `localhost:9000`, nothing is working:
 ```
$ curl 0.0.0.0:9000/api/v1/oauth-clients/local
curl: (56) Recv failure: Connection reset by peer
```
From the container:
Running `netstat -lntu` shows that port 9000 is not open (but port 3000 is open)
```
user@dcb1ba4329cf:~/PeerTube (develop) $ netstat -lntu
Active Internet connections (only servers)
Proto Recv-Q Send-Q Local Address           Foreign Address         State      
tcp        0      0 0.0.0.0:34723           0.0.0.0:*               LISTEN     
tcp        0      0 0.0.0.0:5900            0.0.0.0:*               LISTEN     
tcp        0      0 0.0.0.0:22              0.0.0.0:*               LISTEN     
tcp        0      0 0.0.0.0:3000            0.0.0.0:*               LISTEN     
tcp        0      0 0.0.0.0:8088            0.0.0.0:*               LISTEN     
tcp        0      0 127.0.0.1:5432          0.0.0.0:*               LISTEN     
tcp        0      0 0.0.0.0:8089            0.0.0.0:*               LISTEN     
tcp6       0      0 :::5900                 :::*                    LISTEN     
tcp6       0      0 :::22                   :::*                    LISTEN     
tcp6       0      0 :::8087                 :::*                    LISTEN
```
`curl` also fails:
```
user@dcb1ba4329cf:~/PeerTube (develop) $ curl 0.0.0.0:9000/api/v1/oauth-clients/local
curl: (7) Failed to connect to 0.0.0.0 port 9000: Connection refused
```
#### Now, using my image:
```
docker run -d -i -p 3000:3000 -p 9000:9000 --name peertube my_peertube_dev
docker exec -it peertube bash
(in container) npm run dev
```

Ports 3000 and 9000 are open:
```
root@f587dfeff9cf:/home/user/PeerTube# netstat -tuln
Active Internet connections (only servers)
Proto Recv-Q Send-Q Local Address           Foreign Address         State      
tcp        0      0 127.0.0.1:9000          0.0.0.0:*               LISTEN     
tcp        0      0 0.0.0.0:6379            0.0.0.0:*               LISTEN     
tcp        0      0 0.0.0.0:3000            0.0.0.0:*               LISTEN     
tcp        0      0 127.0.0.1:5432          0.0.0.0:*               LISTEN     
tcp6       0      0 :::6379                 :::*                    LISTEN
```

And `curl` works inside the container:
```
root@f587dfeff9cf:/home/user/PeerTube# curl 0.0.0.0:9000/api/v1/oauth-clients/local
{"client_id":"3fl9b5lb6j4rd6a62311ese9cxq3kcuf","client_secret":"7BLKRtfPAV9RuFRcFuIOeo3UnK9OlsXv"}
```

### Error remaining
Port 9000 seems not reachable from the host, even if it is exposed in the Dockerfile and published to the host. In `localhost:3000`, the web UI still displays `Error getting about from server`, and `curl` fails:
```
am@am:~$ curl 0.0.0.0:9000/api/v1/oauth-clients/local
curl: (56) Recv failure: Connection reset by peer
```

am97 commented:
Some precisions about the Dockerfile:
- I use the `--light` flag after `npm run build` to build only the `en_US` locale (otherwise the build would take too much time).
- `systemctl` command does not exist inside dockers containers ([more info](https://stackoverflow.com/questions/39169403/systemd-and-systemctl-within-ubuntu-docker-images#39169889)), so I use `service postgresql start` to start PostgreSQL.
- For some reason, `service redis-server start` does not work, so I use just `redis-server`

Chocobozzz commented:
If the purpose is to develop with this image, I think you should use `npm run dev` instead of `npm run build` and expose `/home/user/PeerTube` as a volume no?

am97 commented:
Good idea, I edited the Dockerfile to expose `/home/user/PeerTube` as a volume.
`npm run dev` does not work in the Dockerfile, so it's better to run the container and then exec `npm run dev` (I added a usage guide).

I am still not able to access port 9000 from the host, any help is welcome

am97 commented:
I found a workaround to access port 9000: run Firefox inside the Docker container.
I updated the Dockerfile and the [usage guide](https://github.com/am97/PeerTube/blob/new-dockerfile-dev/support/docker/dev/usage.md) accordingly ([this tutorial](http://fabiorehm.com/blog/2014/09/11/running-gui-apps-with-docker/) was of great help).

What do you think?

Chocobozzz commented on support/docker/dev/Dockerfile:
```diff
@@ -1,32 +1,62 @@
+ curl \
+ sudo \
+ nano \
+ nginx \
```
I don't think you this nginx

Chocobozzz commented:
@am97 Please move the old dev docker files in `docker/janitor` directory instead of removing them.

For your server issue, please don't run firefox inside the container. Try with [this patch](https://github.com/Chocobozzz/PeerTube/commit/87bc58e159e305ecada996368deb87376c50647e)

am97 commented:
Moved old dev docker files to `docker/janitor` :heavy_check_mark: 
Removed nginx and firefox from Dockerfile :heavy_check_mark: 
Tested and working :heavy_check_mark: 

Thanks for your help !

Chocobozzz commented:
Thank you @am97!
